src/go/writer/writer.go

```diff
@@ -34,19 +37,39 @@ func (f *featureItem) NodeItem() index.NodeItem {
 	return *f.nodeItem
 }
 
+// writerOption is the additional option to be passed into Writer.
+type writerOption func(*Writer)
+
+// Use memory instead of temporary file when generating index
+// or execute HeaderUpdater.
+// Effective only when includeIndex is true or when HeaderUpdater is provided.
```

Comment:
Maybe expand the comment with a warning that this can use arbitrarily large amounts of memory? fgb files can be *VERY* large. Maybe you even want to add some protection based on the amount of available memory?

Reply:
> Maybe expand the comment with a warning that this can use arbitrarily large amounts of memory?

Ah good point. I will refine the comment.

> Maybe you even want to add some protection based on the amount of available memory?

I think this will require us to use the `runtime` library, which might impact the performance of the flatgeobuf library. I think we can start with a warning message in the comment. We can add the protection later if it is really necessary. WDYT?

Reply:
Works for me.